packages/ibm-products/src/components/UserAvatar/UserAvatar.js

```diff
@@ -46,43 +46,51 @@ const componentName = 'UserAvatar';
  * TODO: A description of the component.
  */
 
-const defaults = {
-  renderIcon: () => <User size={32} />,
-  tooltipAlignment: 'bottom',
-  tooltipText: 'Thomas J. Watson',
-};
-
 export let UserAvatar = React.forwardRef(
   (
     {
       // The component props, in alphabetical order (for consistency).
       backgroundColor,
       className,
+      name,
       /* TODO: add other props for UserAvatar, with default values if needed */
-      renderIcon = defaults.renderIcon,
-      tooltipText = defaults.tooltipText,
-      tooltipAlignment = defaults.tooltipAlignment,
+      renderIcon,
+      size,
+      tooltipText,
+      tooltipAlignment,
       // Collect any other property values passed in.
       ...rest
     },
     ref
   ) => {
     const carbonPrefix = usePrefix();
-    const icons = {
-      user: {
-        md: <User size={32} />,
-      },
-      group: {
-        md: <Group size={32} />,
-      },
+    const iconSize = {
+      sm: 16,
+      md: 20,
+      lg: 24,
+      xl: 32,
+    };
+    const formatInitials = () => {
+      if (name.length === 2) {
```

Comment:
just noticed that we should also tweak this a bit. two problems here

1. when a 2 character string is passed in we're not formatting it with `toUpperCase`
2. if a single string is passed in that's greater than 2, like `David` for example, the output is `DA` which i wouldn't expect because no last name was passed in.

i think we can address this and improve the readability by making some changes
```
const formatInitials = () => {
  const parts = name.split(' ');
  const firstChar = parts[0].charAt(0).toUpperCase();
  if (parts.length === 1) {
    return firstChar;
  }
  const lastChar = parts[parts.length - 1].charAt(0).toUpperCase();
  const initials = [firstChar];
  if (lastChar) {
    initials.push(lastChar);
  }
  return ''.concat(...initials);
};
```

this implementation is slightly more readable than the regex and it clearly demonstrates each outcome

Reply:
Agree with toUpperCase but didn’t mind DA if only two letters are passed. I feel like I’ve seen similar approaches before but unsure if this was intended behavior in our case. 🤔

Reply:
i'll open up an additional issue to confirm this behavior 👍

Reply:
Also found some of the initial discussion (ha ha) that I believe ties to the original implementation. https://github.com/carbon-design-system/ibm-products/issues/69#issuecomment-769895037

It looks like this is to better support use of display names where maybe there is not a first and last name provided or used. Again, probably mostly just a fallback, but I do think we can probably respect the original implementation of this.

(Happy with either the Regex or refactored version of this.)

Reply:
sticked with the refactored version.